### ISSUE TITLE
Phase 4: ML Model Development - Bull/Bear Market Predictor

### ISSUE THREAD

**Issue by kpdagrt22** [kpdagrt22/Astro_Finance_End_to_End_Project#5]
## Objective
Build machine learning models to predict major and minor bull/bear markets based on planetary combinations.

### 1. Market Direction Classifier
**Prediction Target:** Bull/Bear/Neutral market phase

**Models to Try:**
- [ ] Random Forest Classifier
- [ ] XGBoost Classifier
- [ ] LightGBM
- [ ] CatBoost
- [ ] Neural Network (LSTM for time series)
- [ ] Ensemble model (voting/stacking)

**Features:**
- Planetary positions (Rashi/Nakshatra)
- Planetary combinations
- Retrograde indicators
- Dasha periods
- Historical price patterns
- Technical indicators (RSI, MACD, Moving Averages)

### 2. Severity Regressor
**Prediction Target:** Magnitude of bull/bear movement (% change)

**Models to Try:**
- [ ] XGBoost Regressor
- [ ] LightGBM Regressor
- [ ] Random Forest Regressor
- [ ] Gradient Boosting Regressor
- [ ] Neural Network (Dense layers)

### 3. Time Horizon Models
- [ ] **Weekly Predictor** - Next week's market direction
- [ ] **Monthly Predictor** - Next month's market phase
- [ ] **Yearly Predictor** - Annual market outlook

### Data Preparation
- [ ] Feature selection using correlation analysis
- [ ] Handle class imbalance (SMOTE/ADASYN)
- [ ] Train-test split (80-20, time-based)
- [ ] Cross-validation strategy (TimeSeriesSplit)
- [ ] Feature scaling/normalization

### Model Training
```python
# Weekly Model
- Target: market_direction_next_week (Bull/Bear/Neutral)
- Features: 50-100 planetary + technical features
- Evaluation: Accuracy, Precision, Recall, F1-Score

# Monthly Model  
- Target: market_phase_next_month
- Features: Aggregated monthly planetary patterns
- Evaluation: ROC-AUC, Confusion Matrix

# Yearly Model
- Target: annual_market_trend
- Features: Yearly planetary cycles
- Evaluation: Accuracy, Mean Absolute Error
```

### Hyperparameter Tuning
- [ ] Grid Search CV
- [ ] Random Search CV
- [ ] Bayesian Optimization (Optuna)
- [ ] AutoML (TPOT/H2O)

### Feature Importance Analysis
- [ ] SHAP values
- [ ] Permutation importance
- [ ] Feature importance plots
- [ ] Identify top predictive planetary combinations

## Evaluation Metrics
- **Classification:** Accuracy, Precision, Recall, F1-Score, ROC-AUC
- **Regression:** MAE, RMSE, R² Score
- **Business Metrics:** Profitability if traded, Sharpe Ratio

## Model Artifacts
- `models/bull_bear_classifier_weekly.pkl`
- `models/bull_bear_classifier_monthly.pkl`
- `models/bull_bear_classifier_yearly.pkl`
- `models/severity_regressor.pkl`
- `models/feature_importance.json`
- `models/model_performance_report.pdf`

## Validation Strategy
- Walk-forward validation
- Out-of-time testing (2020-2025)
- Backtesting on historical major events

## Success Criteria
- [ ] Weekly model accuracy > 65%
- [ ] Monthly model accuracy > 70%
- [ ] Yearly model accuracy > 75%
- [ ] Successfully predicts major crashes (2008, 2020)
- [ ] Profitable in backtesting

## Dependencies
- Issue #4 (Integrated data)

## Estimated Time
10-14 days

## Priority
🔴 High - Core ML component
